coderd/httpapi/httpapi.go

```diff
@@ -198,6 +199,22 @@ func Write(ctx context.Context, rw http.ResponseWriter, status int, response int
 	_, span := tracing.StartSpan(ctx)
 	defer span.End()
 
+	if rec, ok := rbac.GetAuthzCheckRecorder(ctx); ok {
+		// If you're here because you saw this header in a response, and you're
+		// trying to investigate the code, here are a couple of notable things
+		// for you to know:
+		// - If any of the checks are `false`, they might not represent the whole
+		//   picture. There could be additional checks that weren't performed,
+		//   because processing stopped after the failure.
+		// - The checks are recorded by the `authzRecorder` type, which is
+		//   configured on server startup for development and testing builds.
+		// - If this header is missing from a response, make sure the response is
+		//   being written by calling `httpapi.Write`!
+		// - An empty x-authz-checks header can be valid! Some requests don't
+		//   require authorization.
```

Comment:
It might be no harm to explicitly specify `nil` or something if no authz checks were performed, an empty header has the faint aroma of a bug about it to my nose and could potentially lead someone down the wrong path.

I've also noticed you get an empty `x-authz-check` header back if you hit an authed endpoint without auth. Is this intentional? Would it make sense to set an initial value of the x-authz-checks header that later gets overwritten, so that you can see if there is (no authz) versus (zero authz checks)?